tools/archive/model.py

```diff
@@ -0,0 +1,193 @@
+
+
+    def comments_as_list(self, comments_json):
+        """Create a list of Comments object.
+
+        - author
+        - body
+        - date_created
+        - date_updated
+        """
+        comments = []
+        for comment in comments_json:
+            comments.append(
+                Comment(
+                    author=comment['user']['login'],
+                    body=comment.get('body'),
+                    created_at=to_datetime(comment.get('created_at')),
+                    updated_at=to_datetime(comment.get('updated_at'))
+                    )
+                )
+        return comments
+
+
+    @staticmethod
+    def recursive_fetch(comments_url):
+        """Fetch all comments."""
+        try:
+            r = requests.get(comments_url)
+            r.raise_for_status()
+        except HTTPError as err:
+            logging.warning(f'Fetching comments failed {err}')
+        else:
+            comments = r.json()
+            while 'next' in r.links.keys():
+                r = requests.get(r.links['next']['url'])
+                comments.extend(r.json())
+            return comments
+
+
+@dataclass
+class ArchivedIssue(Issue):
+    """WebCompat Archived Issue Model.
+
+    This provides a couple of additional methods
+    to make it possible to archive the issue as html or any format
+    deemed important. It inherits from a larger Issue model
+    """
+    def as_html(self, template='archive'):
+        """Create an html rendering of an issue.
+
+        template: str
+            directory name with all subtemplates
+        return: str
+            it returns an html file with the content of the issue.
+        """
+        env = Environment(
+            loader=PackageLoader('tools.archive', f'templates/{template}'),
+            autoescape=select_autoescape(['html'])
+        )
+        issue_template = env.get_template('issue.html')
+        if self.locked:
+            locked = 'yes'
+        else:
+            locked = 'no'
+        if self.has_comments():
+            # Let's initialize the comments
+            self.fetch_comments()
+        html_issue = issue_template.render(
+            number = self.number,
+            comments_number = self.comments_number,
+            title = self.title,
+            # TODO: replace self.body markdown
+            issue_body = 'this is a body',
+            locked = locked,
+            issue_created = self.issue_created.strftime('%Y-%m-%d'),
+            issue_updated = self.issue_updated.strftime('%Y-%m-%d'),
+            comments = self.comments,
+            )
+        return html_issue
+
+
+    def save(self, root_dir_path=None):
+        """Save an issue in a specific location.
+
+        root_dir: pathlib.Path
+            full path to where the directory will be saved.
+        return: pathlib.Path
+            it returns the full file location of the archived issue.
+            Probably /static/issue/issue-number.html for webcompat.com
```

Comment:
Just a note for the future, I was wondering if we should hard-code this or make it an environment variable -- we can use nginx to serve the content from a different filesystem location or even an s3 bucket (or whatever, see https://cognition.happycog.com/article/deploying-static-websites-to-aws-s3-behind-an-nginx-proxy, etc).

This could be follow-up work, I don't think we should block on it -- up to you.